File: frontend/src/data/busArrivals.ts
import type { BusStopArrivals } from '@/types';
import busJson from '@data/bus_arrivals.json';

/**
 * 정류장별 버스 도착 — 공유 데이터셋(data/bus_arrivals.json).
 * 저상버스 여부는 true/false/미확인(필드 생략=undefined) 3값으로 구성한다.
 */
export const BUS_ARRIVALS = busJson as unknown as Record<string, BusStopArrivals>;

export const BUS_STOP_LIST = Object.values(BUS_ARRIVALS);

export function getArrivals(stopId: string): BusStopArrivals | undefined {
  return BUS_ARRIVALS[stopId];
}

File: backend/app/data/bus_arrivals.py
"""정류장 저상버스 도착 — 공유 데이터셋(data/bus_arrivals.json)에서 로드."""
from __future__ import annotations

from ..models import BusStopArrivals
from ._loader import load

BUS_ARRIVALS: dict[str, BusStopArrivals] = {
    k: BusStopArrivals.model_validate(v) for k, v in load("bus_arrivals.json").items()
}

BUS_STOP_LIST = list(BUS_ARRIVALS.values())


def get_arrivals(stop_id: str) -> BusStopArrivals | None:
    return BUS_ARRIVALS.get(stop_id)
